internal/scan-manager/scanners/common/src/main/java/org/wso2/security/tools/scanmanager/scanners/common/ScannerController.java

```diff
@@ -96,12 +127,36 @@ public ResponseEntity cancelScan(@RequestBody ScannerScanRequest scanRequest) {
             String message = "No scan running to perform cancellation.";
             responseEntity = new ResponseEntity<>(new ErrorMessage(HttpStatus.NOT_ACCEPTABLE.value(),
                     message), HttpStatus.BAD_REQUEST);
-            log.error(message);
-            CallbackUtil.persistScanLog(scanRequest.getJobId(), message, LogType.ERROR);
         } else {
             log.info("Invoking cancel scan API.");
-            responseEntity = scanner.cancelScan(scanRequest);
+            if (scanner.validateCancelScan(scanRequest)) {
+                if (startScanThread != null) {
+                    startScanThread.interrupt();
+                } else {
+                    log.info("There is no running scan thread to cancel.");
+                }
+
+                cancelScanThread = new Thread(() -> {
```

Comment:
Do we need a reference to `cancelScanThread` at any point. If not can we avoid creation of this variable altogether?